commons-db/src/main/java/com/palantir/nexus/db/DBType.java

```diff
@@ -80,7 +57,7 @@ public static DBType getTypeFromConnection(Connection c) throws PalantirSqlExcep
         if (url.startsWith("jdbc:postgresql")) {
             return POSTGRESQL;
         }
-        if (url.startsWith("jdbc:h2:mem:")) {
```

Comment:
Did this break the tests? Seems like we would still expect in-memory DBs for H2 to have this prefix - [H2 docs](http://www.h2database.com/html/features.html#in_memory_databases).

Reply:
The question is whether we want to support non-in-memory H2. The connection config allows you to override the URL to point to a file on disk instead of in-memory.

So we should either:
- Make the URL non-configurable
- Update this logic to correctly handle non-in-memory H2

Reply:
Although I guess you can support multiple in-memory databases by using names in the URL (ex. `jdbc:h2:mem:db1`). So maybe we want to just leave that configurable.